app/scripts/lib/channels/receivers/web-channel.js

```diff
@@ -48,12 +49,51 @@ define(function (require, exports, module) {
       // https://bugzilla.mozilla.org/show_bug.cgi?id=1173830
       //
       // Ignore events with no `message` field.
-      var message = detail.message;
-      if (message) {
+      let message = detail.message;
+      if (message && ! this._reportedError(message)) {
         this.trigger('message', message);
       }
     },
 
+    /**
+     * Determine if the message received had an error reported.
+     *
+     * @param {Object} message received from the WebChannel
+     * @returns {Boolean}
+     * @private
+     */
+    _reportedError (message) {
+      // this is super confusing, so read carefully:
+      // there are two ways the error can be reported. Either `message.error` or `message.data.error`.
+      let errorMsg = 'Unknown error';
+      let errorStack = null;
+      let reportedError = false;
+
+      if (message.error && _.isString(message.error)) {
+        // if it is a String then it is probably an error from WebChannel.jsm
+        // Example: https://dxr.mozilla.org/mozilla-central/rev/bad312aefb42982f492ad2cf36f4c6c3d698f4f7/toolkit/modules/WebChannel.jsm#101
+        errorMsg = message.error;
+        reportedError = true;
+      } else if (message.data && message.data.error) {
+        // if it has an error Object that means it is a component error with a stack
+        errorMsg = message.data.error.message;
+        errorStack = message.data.error.stack;
+        reportedError = true;
+      }
+
+      if (reportedError) {
+        this._logger.error('WebChannel error:', errorMsg);
```

Comment:
Two comments about this section:

1. Reporting the error is a pretty big side effect when the function name implies the function only checks whether the message contains an error. Can the reporting logic be moved out of here into `receiveMessage` or the function renamed?
2. Before propagating the error message/stack, should the existing of the `errorMsg` and `errorStack` be done?
3. It looks like it's possible for `errorStack` to be `null` if the `if (message.error && _.isString(message.error))` is taken, will that cause any issues on Raven?